ctapipe/analysis/tests/test_sensitivity.py

```diff
@@ -0,0 +1,219 @@
+    Noff = 10
+    energy_bin_edges = np.array([.1, 10])*u.TeV  # one bin with the bin-centre at 1 TeV
+    energies_sig = np.ones(Non)
+    energies_bgr = np.ones(Noff)
+
+    sens = SensitivityPointSource(
+            {'s': energies_sig*u.TeV,   # all events have 1 TeV energy
+             'b': energies_bgr*u.TeV})  # all events have 1 TeV energy
+
+    sensitivity = sens.get_sensitivity(alpha=alpha, signal_list=("s"), mode="data",
+                                       sensitivity_source_flux=crab_source_rate,
+                                       sensitivity_energy_bin_edges=energy_bin_edges)
+
+    # the ratio between sensitivity and reference flux (i.e. from the crab nebula) should
+    # be the scaling factor that needs to be applied on Nsig to produce a 5 sigma result
+    # in the lima formula
+    ratio = sensitivity["Sensitivity"][0]/(crab_source_rate(1*u.TeV)).value
+    np.testing.assert_allclose([sigma_lima(ratio*(Non-Noff*alpha)+Noff,
+                                           Noff, alpha)], [5])
+
+
+def test_generate_toy_timestamps():
+    from scipy import signal
+    gamma_light_curve = signal.gaussian(5000, 700) * 1000
+
+    tmin, tmax = 0, 5
+
+    Nthrows = 500
+
+    time_stamps = SensitivityPointSource.\
+        generate_toy_timestamps(light_curves={'g': gamma_light_curve,
+                                              'p': Nthrows},
+                                time_window=(tmin, tmax))
+
+    assert len(time_stamps['g']) == int(np.sum(gamma_light_curve))
+    assert len(time_stamps['p']) == Nthrows
+    for st in time_stamps.values():
+        assert (np.min(st) >= tmin) and (np.max(st) <= tmax)
+        np.testing.assert_allclose(np.mean(st), 2.5, atol=.25)
+
+
+def test_draw_events_with_flux_weight():
+
+    Emin = 20
+    Emax = 50
+    Nbin = 50
+    Ndraws = 10000
+
+    def weight_from_energy(e, gamma_new=3, gamma_old=2):
+        """
+        use this to determine event weights (i.e. probabilities to get picked in the
+        random draw
+
+        Parameters
+        ----------
+        e : numpy array
+            the energies of the MC events
+        gamma_new : float
+            the spectral index the drawn set of events are supposed to follow
+        gamma_old : float
+            the spectral index the MC events have been generated with
+
+        Returns
+        -------
+        weights : numpy array
+            list of event weights normalised to 1
+            to be used as PDF in `np.random.choice`
+        """
+        e_w = e**(gamma_old - gamma_new)
+        return e_w / np.sum(e_w)
+
+    energy_sel_gamma = np.random.uniform(Emin, Emax, 50000)
+
+    indices = SensitivityPointSource.draw_events_from_flux_weight(
+                        {'g': energy_sel_gamma},
+                        {'g': weight_from_energy(energy_sel_gamma, 2, 0)},
+                        {'g': Ndraws})
+
+    assert len(energy_sel_gamma[indices['g']]) == Ndraws
+    # TODO come up with more tests...
```

Comment:
I'd suggest to always seed the RNG if you have tests that use RNG, so that results are always the same.
(tests where the results change on every run are painful to maintain and debug.)

Then when you always have the same result, you can actually assert on some number from your output, not just have these high-level consistency checks. In my experience there are always bugs and changes in method and code refactorings, and having tests with asserts on the actual results that come out of the functions is useful (even if now just establishing the status quo, not somehow "verified" numbers)